### PR TITLE
Implements transfer server and user

### LINKED ISSUE #722: Implement AWS Transfer
### What problem are you facing?
Amazon Transfer resources are missing.

https://pkg.go.dev/github.com/aws/aws-sdk-go@v1.38.61/service/transfer

### How could Crossplane help solve your problem?
Use ACK to generate transfer and then fill the gaps.

### PR DESCRIPTION
**Pull request by haarchri**
Signed-off-by: Christopher Haar <chhaar30@googlemail.com>

### Description of your changes
implements transfer 

- server
- user

Fixes #722 #748

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

```
kubectl get managed
NAME                                        READY   SYNCED   EXTERNAL-NAME
server.transfer.aws.crossplane.io/example   True    True     s-c1ed444984374e53a

NAME                                      READY   SYNCED   EXTERNAL-NAME
user.transfer.aws.crossplane.io/example   True    True     example
```

```
kubectl describe server.transfer.aws.crossplane.io/example
Name:         example
Namespace:    
Labels:       <none>
Annotations:  crossplane.io/external-name: s-c1ed444984374e53a
API Version:  transfer.aws.crossplane.io/v1alpha1
Kind:         Server
Metadata:
  Creation Timestamp:  2021-07-05T20:05:18Z
  Finalizers:
    finalizer.managedresource.crossplane.io
  Generation:        2
  Resource Version:  1269563
  Self Link:         /apis/transfer.aws.crossplane.io/v1alpha1/servers/example
  UID:               95dd807d-c4da-499c-ba60-756fa23130c3
Spec:
  Deletion Policy:  Delete
  For Provider:
    Domain:  S3
    Endpoint Details:
      Security Group ID Refs:
        Name:  example
      Security Group I Ds:
        sg-0858486fe394890bf
      Subnet ID Refs:
        Name:  sample-subnet1
      Subnet I Ds:
        subnet-f2ccb898
      Vpc ID:  vpc-aab73bc0
      Vpc ID Ref:
        Name:                sample-vpc
    Endpoint Type:           VPC
    Identity Provider Type:  SERVICE_MANAGED
    Protocols:
      SFTP
    Region:                eu-central-1
    Security Policy Name:  TransferSecurityPolicy-2020-06
    Tags:
      Key:    myKey
      Value:  myValue
  Provider Config Ref:
    Name:  default
  Write Connection Secret To Ref:
    Name:       example-conn
    Namespace:  crossplane-system
Status:
  At Provider:
    Server ID:  s-c1ed444984374e53a
  Conditions:
    Last Transition Time:  2021-07-05T20:05:20Z
    Reason:                ReconcileSuccess
    Status:                True
    Type:                  Synced
    Last Transition Time:  2021-07-05T20:06:21Z
    Reason:                Available
    Status:                True
    Type:                  Ready
Events:
  Type    Reason                   Age   From                                       Message
  ----    ------                   ----  ----                                       -------
  Normal  CreatedExternalResource  69s   managed/server.transfer.aws.crossplane.io  Successfully requested creation of external resource
```

```
kubectl describe user.transfer.aws.crossplane.io/example
Name:         example
Namespace:    
Labels:       <none>
Annotations:  crossplane.io/external-name: example
API Version:  transfer.aws.crossplane.io/v1alpha1
Kind:         User
Metadata:
  Creation Timestamp:  2021-07-05T20:06:38Z
  Finalizers:
    finalizer.managedresource.crossplane.io
  Generation:        2
  Resource Version:  1269634
  Self Link:         /apis/transfer.aws.crossplane.io/v1alpha1/users/example
  UID:               f9fc8845-dfbe-48fa-832c-82c51b3aa164
Spec:
  Deletion Policy:  Delete
  For Provider:
    Home Directory:       /aws-glue-crossplane/example
    Home Directory Type:  PATH
    Region:               eu-central-1
    Role:                 arn:aws:iam::1234567891012:role/aws-glue
    Role Ref:
      Name:     aws-glue
    Server ID:  s-c1ed444984374e53a
    Server ID Ref:
      Name:               example
    Ssh Public Key Body:  ssh-rsa <ssh-key>
    Tags:
      Key:    myKey
      Value:  myValue
  Provider Config Ref:
    Name:  default
Status:
  At Provider:
    Server ID:  s-c1ed444984374e53a
  Conditions:
    Last Transition Time:  2021-07-05T20:06:39Z
    Reason:                ReconcileSuccess
    Status:                True
    Type:                  Synced
    Last Transition Time:  2021-07-05T20:06:39Z
    Reason:                Available
    Status:                True
    Type:                  Ready
Events:
  Type    Reason                   Age    From                                     Message
  ----    ------                   ----   ----                                     -------
  Normal  CreatedExternalResource  2m14s  managed/user.transfer.aws.crossplane.io  Successfully requested creation of external resource
```

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: examples/transfer/server.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+kind: Server
+metadata:
+  name: example
+spec:
+  forProvider:
+    region: us-east-1
+    domain: S3
+    endpointDetails:
+      securityGroupIDRefs:
+        - name: example
```
[the remark this answers is not in the document]
Comment (haarchri) [reply examples/transfer/server.yaml]: changed the example, squashed commits & rebase master 
thanks for review ;)

**Comment by haarchri** [comment]
changed the example, squashed commits & rebase master
thanks for review ;)

**Comment by haarchri** [comment]
@AaronME is something missing - or can we approve and merge it ?

**Comment by haarchri** [comment]
@AaronME  can you run CI Job ? Something missing here ?

**Comment by tiegs** [comment]
What is the status of this feature? We were hoping to have this as part of the last release - can you provide any ETA on this?

**Review by AaronME — APPROVED** [review]
@haarchri This looks great! Thank you for the contribution.

I was able to create and delete the examples with this code.

**Pull request merged**